.github/workflows/main.yml

```diff
@@ -309,3 +309,806 @@ jobs:
+          restore-keys: ${{ runner.os }}-
+
+      - name: Check
+        run: ant verify-libs-and-licenses -Dverify-libs-and-licenses.haltonfailure=true -Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json
+
+  run-sigtests:
+    name: Run sigtests
+    runs-on: ubuntu-18.04
+    env:
+      OPTS: -Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json -quiet -Djavac.compilerargs=-nowarn -Dbuild.compiler.deprecation=false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Caching dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.hgexternalcache
+          key: ${{ runner.os }}-${{ hashFiles('**/external/binaries-list') }}
+          restore-keys: ${{ runner.os }}-
+
+      - name: Clean
+        run: ant $OPTS clean
+
+      - name: Build
+        run: ant $OPTS build
+
+      - name: Check sigtests
+        run: ant $OPTS check-sigtests-release -Dfail.on.error=true
+
+      - name: Generate sigtests release
+        run: ant $OPTS gen-sigtests-release
+
+  run-sigtests-platform:
+    name: Run gen-sigtests-release for platform cluster
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Caching dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.hgexternalcache
+          key: ${{ runner.os }}-${{ hashFiles('**/external/binaries-list') }}
+          restore-keys: ${{ runner.os }}-
+
+      - name: Gen sigtests
+        run: nbbuild/travis/gensigtest.sh
+
+  pristine-platform-build:
+    name: Pristine platform build
+    runs-on: ubuntu-18.04
+    env:
+      ANT_OPTS: -Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json -Djavac.compilerargs=-nowarn -Dbuild.compiler.deprecation=false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Caching dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.hgexternalcache
+          key: ${{ runner.os }}-${{ hashFiles('**/external/binaries-list') }}
+          restore-keys: ${{ runner.os }}-
+
+      - name: Build
+        run: |
+          ant -quiet build-source-config -Dcluster.config=platform
+          mkdir tmpplatform
+          cd tmpplatform
+          unzip -qq ../nbbuild/build/platform-src*
+          ant build -Dcluster.config=platform
+
+  compile-all-modules-openjdk8:
+    name: Compile all modules with OpenJDK8
+    runs-on: ubuntu-18.04
+    env:
+      OPTS: -Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json -quiet -Dcluster.config=full -Djavac.compilerargs=-nowarn -Dbuild.compiler.deprecation=false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Caching dependencies
```

Comment:
Add "setup-java" step for Java8 rather than relying on Java8 to be the default for the Runner. (making it explicit rather than implied)